src/analysis/state_helper.rs

```diff
@@ -63,6 +63,11 @@ fn find_previous_non_empty_split_and_comparison_time(
 /// Calculates the comparison's segment time of the segment with the timing
 /// method specified. This is not calculating the current attempt's segment
 /// times.
+///
+/// # Panics
+///
+/// Panics if the provided `segment_index` is greater than or equal to
+/// `run.len()`.
 pub fn comparison_segment_time(
     run: &Run,
     segment_index: usize,
```

Comment:
This function should have the additional checks that the detailed timer is doing, such as the specialized logic for the best segments comparison and so on. The goal should be to remove the function from the detailed timer and use this one instead. Then it should be good to merge.

Reply:
@wooferzfg Looks like in the splits we use "combined comparison segment time", in the detailed timer we use "single comparison segment time" and for the segment time component it could potentially be either. What do you think we should do? Do the same thing for all of them?

Reply:
My initial reasoning for using the combined segment time in the splits vs. a single segment time in the detailed timer is that you can see all of the segments in the splits, which means you know which segment the combined segment time started at. In the detailed timer, you can only see the current segment, so we never show combined segments. There's also the issue of the segment timer starting at a different segment than the comparison segment time if we start combining stuff.

However, we don't really apply this logic everywhere else, so it's not totally clear what we should do here. The previous segment component uses combined segments, for example. In my opinion, it would be best to use the combined comparison segment time here because there's only one piece of information being displayed.

Reply:
Yeah I think for the splits it makes sense to keep it showing the combined segments, as you can see that the ones before are skipped. For the detailed timer you don't, so it makes sense to dash it out. But I'm not quite following your argument for the segment time component, as you seem to argue the opposite there?
> combined comparison segment time here because there's only one piece of information being displayed.

Reply:
For the segment time component, we only show one value, so there's no issue with conflicting information within the component (unlike the detailed timer, where we show a segment timer and 2 different comparisons). I don't feel too strongly about this though.

Reply:
I don't have many strong feelings here either, but if I was forced to choose I'd say just unify them for the sake of code quality and consistency.